src/Jackett.Common/Indexers/BJShare.cs

```diff
@@ -520,7 +531,17 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                             // release.Title += string.Join(" ", titleElements);
                             release.Title = release.Title.Trim();
 
-                            release.Title += " " + titleElements[5] + " " + titleElements[3] + " " + titleElements[1] + " " + titleElements[2] + " " + titleElements[4] + " " + string.Join(" ", titleElements.Skip(6).Take(titleElements.Length - 6).ToArray());
+                            if (titleElements.Length < 6)
+                            {
+                                // Usually non movies / series could have less than 6 elements, eg: Books.
+                                release.Title += " " + string.Join(" ", titleElements.Take(titleElements.Length).ToArray());
```

Comment:
No need to use `Take()` when getting the whole list, nor need to call `ToArray()` as `string.Join` accepts any `IEnumerable` type. 

```suggestion
                                release.Title += " " + string.Join(" ", titleElements);
```